isisd/isis_northbound.c

```diff
@@ -1702,6 +1704,42 @@ static int lib_interface_isis_ipv6_routing_modify(enum nb_event event,
 	return NB_OK;
 }
 
+/*
+ * XPath: /frr-interface:lib/interface/frr-isisd:isis/bfd-monitoring
+ */
+static int lib_interface_isis_bfd_monitoring_modify(enum nb_event event,
+						    const struct lyd_node *dnode,
+						    union nb_resource *resource)
+{
+	struct isis_circuit *circuit;
+	bool bfd_monitoring;
+
+	if (event != NB_EV_APPLY)
+		return NB_OK;
+
+	circuit = nb_running_get_entry(dnode, NULL, true);
+	bfd_monitoring = yang_dnode_get_bool(dnode, NULL);
+
+	if (bfd_monitoring) {
+		/* Sanity check: BFD session is already registered. */
+		if (circuit->bfd_info != NULL)
+			return NB_OK;
```

Comment:
I think these sanity checks are unnecessary since this callback won't be called unless the value of `bfd-monitoring` has changed.

If you want to preserve the sanity checks nonetheless, please return `NB_ERR_INCONSISTENCY` instead so that the northbound code will log the internal inconsistency.